sdks/java/testing/jpms-tests/build.gradle

```diff
@@ -33,30 +40,15 @@ enableJavaPerformanceTesting()
 description = "Apache Beam :: SDKs :: Java :: Testing :: JPMS Tests"
 ext.summary = "E2E test for Java 9 modules"
 
-// Java 17 needs compileJava to add-exports and add-opens for error prone
-if (project.hasProperty("compileAndRunTestsWithJava17")) {
-  def java17Home = project.findProperty("java17Home")
-  project.tasks.withType(JavaCompile) {
+// direct compileJava to use specified java version.
+project.tasks.compileJava {
```

Comment:
This makes both (default JDK is Java9+) and (default JDK is Java8 and pass `java11Home` or `java17Home`) works.

Reply:
Seems OK to me if it works. Do you need `setJava11Options`?

Reply:
We do not have `setJava11Options` or correspondents in beamModulePlugin

setJava17Options is due to new module export rules needed in Java17 (https://blogs.oracle.com/javamagazine/post/a-peek-into-java-17-continuing-the-drive-to-encapsulate-the-java-runtime-internals), it does the same thing as the current code (removed/simplified in this PR).